src/test/java/net/minecraftforge/debug/misc/DataPackRegistriesTest.java

```diff
@@ -0,0 +1,228 @@
+                () -> new RegistryBuilder<Unsyncable>().disableSaving().dataPackRegistry(Unsyncable.DIRECT_CODEC));
+            // The overload of #dataPackRegistry that takes a second codec marks the datapack registry as syncable.
+            syncables.makeRegistry(Syncable.class,
+                () -> new RegistryBuilder<Syncable>().disableSaving().dataPackRegistry(Syncable.DIRECT_CODEC, Syncable.DIRECT_CODEC));
+            
+            // Datapack registry elements can be datagenerated, but they must be registered as builtin objects first.
+            this.datagenTestObject = unsyncables.register("datagen_test", () -> new Unsyncable("Datagen Success"));
+            
+            unsyncables.register(modBus);
+            syncables.register(modBus);
+            
+            modBus.addListener(this::onGatherData);
+            forgeBus.addListener(this::onServerStarting);            
+            
+            if (FMLEnvironment.dist == Dist.CLIENT)
+            {
+                ClientEvents.subscribeClientEvents();
+            }
+        }
+    }
+    
+    private void onGatherData(final GatherDataEvent event)
+    {
+        if (!event.includeServer())
+            return;
+        // Example of how to datagen datapack registry objects.
+        // Objects to be datagenerated must be registered (e.g. via DeferredRegister above).
+        // This outputs to data/data_pack_registries_test/data_pack_registries_test/unsyncable/datagen_test.json
+        final DataGenerator generator = event.getGenerator();
+        final Path outputFolder = generator.getOutputFolder();
+        final RegistryOps<JsonElement> ops = RegistryOps.create(JsonOps.INSTANCE, RegistryAccess.BUILTIN.get());
+        final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        final ResourceLocation registryId = Unsyncable.REGISTRY_KEY.location();
+        final ResourceLocation id = this.datagenTestObject.getId();
+        final Unsyncable element = this.datagenTestObject.get();
+        final String pathString = String.join("/", PackType.SERVER_DATA.getDirectory(), id.getNamespace(), registryId.getNamespace(), registryId.getPath(), id.getPath()+".json");
+        final Path path = outputFolder.resolve(pathString);
+        generator.addProvider(new DataProvider()
+        {
+            @Override
+            public void run(final HashCache cache) throws IOException
+            {
+                Unsyncable.DIRECT_CODEC.encodeStart(ops, element)
+                    .resultOrPartial(msg -> LOGGER.error("Failed to encode {}: {}", path, msg)) // log error on encode failure
+                    .ifPresent(json -> // output to file on encode success
+                    {
+                        try
+                        {
+                            DataProvider.save(gson, cache, json, path);
+                        }
+                        catch(IOException e) // we're inside this ifpresent so the throws can't deal with this
```

Comment:
Formatting: space after `catch` keyword.
```suggestion
                        catch (IOException e) // we're inside this ifpresent so the throws can't deal with this
```
Also, perhaps you mean `as` ("as the throws can't deal with this") and not `so`? Unless I'm interpreting the comment's intention incorrectly, the reason the `catch` exists is because we cannot propagate the `IOException` upwards normally because of the use of a lambda, so `as` would be more semantically correct here.

Reply:
You're interpreting it correctly, but I don't think that would be a correct use of "as" either. I'll use "because" for minimum ambiguity -- "The throws can't deal with this exception, because we're inside this ifPresent."